### ISSUE TITLE
FailOpen lock probably won't work if heartbeatPeriodMs is unspecified. Also, attempting to release the lock twice is unsafe.

### ISSUE THREAD

**Issue by tristanls** [tristanls/dynamodb-lock-client#22]
Looking at [this `if` statement](https://github.com/tristanls/dynamodb-lock-client/blob/master/index.js#L403) :
```javascript
...
    if (self._heartbeatTimeout)
    {
        clearTimeout(self._heartbeatTimeout);
        return self._releaseFailOpen(callback);
    }
    else
    {
        return self._releaseFailClosed(callback);
    }
...
```
`self._heartbeatTimeout` is used as an indirect way to determine whether the lock is fail open or closed. 

Since `heartbeatPeriodMs` is optional, it can be left `undefined`. If that's the case, then `self._heartbeatTimeout` will not be set because [this `if` statement](https://github.com/tristanls/dynamodb-lock-client/blob/master/index.js#L349) is bypassed when `heartbeatPeriodMs` is falsy and the code will erronously try to release a fail open lock using fail closed mechanism.

**Comment by tristanls** [comment]
The other use case to update this code is being able to retry lock release if it fails, as described in #21 .

**Comment by tristanls** [comment]
This should be resolved with [v0.6.2](https://github.com/tristanls/dynamodb-lock-client/releases/tag/v0.6.2) release.
